kyaml/openapi/openapi.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package openapi
+
+import (
+	"sync"
+
+	"github.com/go-openapi/spec"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var setup sync.Once
+var schema spec.Schema
+var schemaByResourceType map[yaml.TypeMeta]*spec.Schema
+
+func initSchema() {
+	setup.Do(func() {
+		schemaByResourceType = map[yaml.TypeMeta]*spec.Schema{}
+
+		// parse the swagger, this should never fail
+		b := MustAsset("openapi/swagger.json")
+		if err := schema.UnmarshalJSON(b); err != nil {
+			panic(err)
+		}
+		for k := range schema.Definitions {
+			d := schema.Definitions[k]
+			gvk, found := d.VendorExtensible.Extensions["x-kubernetes-group-version-kind"]
+			if !found {
+				continue
+			}
+			exts, ok := gvk.([]interface{})
+			if !ok || len(exts) != 1 {
+				continue
+			}
+			m, ok := exts[0].(map[string]interface{})
+			if !ok {
+				continue
+			}
+			g := m["group"].(string)
+			apiVersion := m["version"].(string)
+			if g != "" {
+				apiVersion = g + "/" + apiVersion
+			}
+
+			schemaByResourceType[yaml.TypeMeta{Kind: m["kind"].(string), APIVersion: apiVersion}] = &d
```

Comment:
I see several dispersed raw strings in code(kind, apiVersion, version etc.). As these are standard limited kubernetes yaml key terms, I would suggest having a them in a separate file as constants and refer them all across the code base. This will avoid typos and increases maintainability of code.

Reply:
done